From 67946055a012137375d15d1b9429b9ab53898d60 Mon Sep 17 00:00:00 2001
From: juntingwang <Junting.Wang@windriver.com>
Date: Fri, 1 Mar 2013 16:02:56 +0800
Subject: Set PS2 Keyboard controller as not a keyboard type

The Eventhub detects the platforms built-in PS/2 keyboard controller
as a physical keyboard. So the system will not display the software
keyboard by default. Modify it device type in Keyboard-less tablets,
so that ActivityManager will not ignore the PS/2 controller.

Category: Device Enablement
Domain: IO
Upstream-Candidate: yes

Issue: AXIA-1875

Change-Id: Ia5b7592e82aaf1970066bab5e9ca5f46ac83ab1c
Signed-off-by: juntingwang <Junting.Wang@windriver.com>
---
 services/input/EventHub.cpp |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/services/input/EventHub.cpp b/services/input/EventHub.cpp
index 4d70d5f..a91218d 100644
--- a/services/input/EventHub.cpp
+++ b/services/input/EventHub.cpp
@@ -1197,7 +1197,11 @@ status_t EventHub::openDeviceLocked(const char *devicePath) {
 
         // 'Q' key support = cheap test of whether this is an alpha-capable kbd
         if (hasKeycodeLocked(device, AKEYCODE_Q)) {
-            device->classes |= INPUT_DEVICE_CLASS_ALPHAKEY;
+            char value[PROPERTY_VALUE_MAX];
+            property_get("ro.ignore_atkbd", value, "0");
+            if ((device->identifier.name !="AT Translated Set 2 keyboard") || (!atoi(value))) {
+                device->classes |= INPUT_DEVICE_CLASS_ALPHAKEY;
+            }
         }
 
         // See if this device has a DPAD.
-- 
1.7.4.1

